src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -347,17 +430,43 @@ private static Details getDetails(DetailAST ast) {
                     shouldCheckLastRcurly = true;
                     nextToken = getNextToken(ast);
                 }
-                break;
-            case TokenTypes.LITERAL_CATCH:
+            }
+            else if (tokenType == TokenTypes.LITERAL_CATCH) {
                 nextToken = ast.getNextSibling();
                 lcurly = ast.getLastChild();
                 rcurly = lcurly.getLastChild();
                 if (nextToken == null) {
                     shouldCheckLastRcurly = true;
                     nextToken = getNextToken(ast);
                 }
-                break;
-            case TokenTypes.LITERAL_IF:
+
+            }
+            else {
+                shouldCheckLastRcurly = true;
+                nextToken = getNextToken(ast);
+                lcurly = ast.getFirstChild();
+                rcurly = lcurly.getLastChild();
+            }
+            final Details details = new Details();
```

Comment:
I like the separation the new way gives for each method, but I dislike some of the code it has to copy between the 2.

I am not sure about which of 2 ideas to change this to:
1)  How about we drop the default constructor and make a new one that takes all fields as arguments, then we could make the fields of the class `final`?
2) Why don't we transform `getDetails` into a constructor, and have all these methods populate the fields directly.

Problem with point 2 is assignments could be overlooked since they are split into so many methods.
Problem with point 1 is we are stuck with `getDetails` as a wrapper for the constructor.

What do you all think?